csrc/scheduler/vectorize_helper.cpp

```diff
@@ -843,6 +844,95 @@ std::vector<std::unordered_map<TensorView*, Val*>> getTvToContigInnerSizeMapsOf(
   return mappers;
 }
 
+// This is a WAR for vectorizing through resized iter domains. The
+// spanning tree based analysis is not guaranteed to take all resize
+// ops into considerations (issue
+// https://github.com/NVIDIA/Fuser/issues/3640). To workaround the
+// limitation, grab all factors that must be divisible by a
+// vectorization factors.
+std::unordered_set<Val*> getResizeVectorizationFactors(
+    TensorView* reference_tv,
+    int64_t break_point) {
+  Fusion* fusion = reference_tv->fusion();
+  std::unordered_set<Val*> factors;
+  const auto resize_based_ops = scheduler_tools::getResizeBasedOps(fusion);
+
+  if (resize_based_ops.empty()) {
+    return factors;
+  }
+
+  IdModel id_model(reference_tv->fusion());
+  const auto& graph = id_model.buildExactGraph();
+
+  const auto ref_groups = graph.toGroups(reference_tv->getLogicalDomain());
+
+  // For each of resize-based tensor ops, find all resize ops
+  // that exist between the vectorized reference IDs and the output
+  // tensor.
+  for (auto resize_based_op : resize_based_ops) {
+    auto resize_out = resize_based_op->output(0)->as<TensorView>();
+    NVF_ERROR(
+        resize_out->hasRoot(), "Unexpected op: ", resize_based_op->toString());
+    // To make sure the resize op of this resize_based_op tensor op,
+    // use both the root and logical domains as the traversal targets
+    ValGroups resize_inp_out;
+    resize_inp_out.pushBack(graph.toGroups(resize_out->getRootDomain()));
+    resize_inp_out.pushBack(graph.toGroups(resize_out->getLogicalDomain()));
```

Comment:
Is the reason of pushing both root and logical because we could be traversing on both directions?

If so, I think the comment on line 876-877 isn't very clear about that.

Reply:
That is correct. Updated the comment.